src/get_programming_with_js_next/module_7_exercise.es6

```diff
@@ -0,0 +1,48 @@
+/*
+  Write a function that you can use to tag template literals that checks each
+  value if it is an object, and if so, converts the object into a string of
+  key/value pairs before interpolation.
+
+  Input:
+
+  const props = {
+    src: 'http://fillmurray.com/100/100',
+    alt: 'Bill Murray'
+  };
+
+  const img = withProps`<img ${props}>`;
+
+  Output:
+
+  <img src="http://fillmurray.com/100/100" alt="Bill Murray">
+*/
+
+const PROPS = {
+  src: 'http://fillmurray.com/100/100',
+  alt: 'Bill Murray'
+};
+
+function formatAttributes(attributes) {
+  let formattedAttributes = '';
+
+  for (let attribute in attributes) {
+    formattedAttributes += `${attribute}="${attributes[attribute]}" `;
```

Comment:
Guard your properties when iterating across object keys, as prototype issues can cause problems: https://eslint.org/docs/rules/guard-for-in

Also, to avoid mutation (not a requirement, just a note if we go fully immutable - which we can't yet) you can use fancy array shit here:
```javascript
function formatAttributes(attributes) {
  return Object.keys(attributes).reduce(function stringifyAttr(str, attr) {
    return `${str} ${attr}="${attributes[attr]}"`;
  }, '').trim();
};
```

Reply:
Wasn't aware of this. I'm going to add ESLint to the build script. Does my change work?

Reply:
👍